combine/combine_test.go

```diff
@@ -0,0 +1,276 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Genome Research Ltd.
+ *
+ * Author: Sendu Bala <sb10@sanger.ac.uk>
+ * 		   Kyle Mace  <km34@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package combine
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/wtsi-ssg/wrstat/v3/fs"
+)
+
+// TestConcatenateMergeAndCompress tests the concat, merge, and compress
+// functionality of the combine package.
+func TestConcatenateMergeAndCompress(t *testing.T) {
+	Convey("Given some inputs and an output", t, func() {
+		inputs, output, outputPath := BuildCombineAndUserGroupInputs(t,
+			[]string{"path1", "path2", "path3", "path4", "path5", "path6"},
+			[]string{"adam", "ben", "charlie"},
+			[]string{"david", "fred", "graham"},
+			[]string{"short/test/dir/", "test/dir/that/is/much/longer"})
```

Comment:
Can we have a test with a dir that is the maximum file path length (4096 characters)?